apollo-api/src/main/java/com/spotify/apollo/RequestValue.java

```diff
@@ -88,7 +88,9 @@ public Request withHeader(String name, String value) {
   @Override
   public Request withHeaders(Map<String, String> additionalHeaders) {
     Map<String, String> headers = new LinkedHashMap<>(headers());
-    headers.putAll(additionalHeaders);
+    for (Map.Entry<String, String> header : additionalHeaders.entrySet()) {
+      headers.put(header.getKey().toLowerCase(), header.getValue());
+    }
```

Comment:
Can Apollo as it stands even handle protocols other than HTTP? It seems to be very much built up around HTTP and the lifecycle of an HTTP request.